src/main/java/com/github/manolo8/darkbot/core/manager/MapManager.java

```diff
@@ -421,7 +421,9 @@ private Location getEnemyLocatorTarget() {
     }
 
     private boolean findMarker(long spriteArray, double scale, Location result) {
-        int size = API.readMemoryInt(spriteArray, 0x40, 0x18);
+        long array_data = API.readMemoryLong(spriteArray, 0x40);
+        if (array_data == 0 ) return false;
+        int size = API.readMemoryInt(array_data + 0x20);
```

Comment:
these are equivalent except for the 0x18 -> 0x20 offset, readMemoryLong checks at every step of the way the ptr is valid now